python/mpi_cmake_modules/clang_format.py

```diff
@@ -50,7 +50,25 @@ def test_valid_file(filename):
         return False
 
 
+# either returns file_or_directory as such (if exists)
+# or assumes file_or_directory was a relative path and returns
+# the corresponding absolute path (if exist, None otherwise)
+def _get_full_path(file_or_directory):
+    if path.isfile(file_or_directory) or path.isdir(file_or_directory):
+        return file_or_directory
+    fixed_path = os.path.abspath(os.getcwd()+os.sep+file_or_directory)
```

Comment:
You should use os.path.join(os.getcwd(), file_or_directory)
I am also a little confused about using os.path.abspath + os.getcwd()...
Just os.path.abspath should suffice no?

Reply:
- using join would lead to the same result.
- I do not understand second comment. os.getcwd() is what gets the current directory, so it is needed.

Reply:
abspath uses getcwd

Reply:
just tested, you are right. Did not know that :) Will fix.